### DISCUSSION TITLE
SFTP 如何使用密钥登录

### DISCUSSION THREAD

**Discussion by loveqianool** [devld/go-drive#42] [Q&A] [answered]
我直接将密钥内容填入 HostKey 不行，尝试填入密钥文件的路径也不行，应该怎么填写呢？
服务器端禁用了密码登录，而在这里密码是必填的，是不是优先使用密码登录了呢？

**Comment by devld** [comment]
密钥登录还没支持，这两天我支持一下。

HostKey 是用于确认远程主机身份的，一般可以不填。

**Comment by devld** [comment]
[`v0.6.4`](https://github.com/devld/go-drive/releases) 已支持 SFTP 私钥认证，但目前只支持 RSA 私钥，并且这个私钥不能是有密码保护的。

**Comment by loveqianool** [comment]
好用了，成功连接上了，谢谢大佬。
另外我还发现了两个问题
我在配置文件的 type: text 的 file-types 内加入了 conf，但它没有生效，我仍然无法编辑 nginx 的配置文件。
还有在添加驱动器的时候，驱动器的名称不能有数字零，不知道这是设计的如此还是 bug。

**Reply by devld** [reply]
> 好用了，成功连接上了，谢谢大佬。 另外我还发现了两个问题 我在配置文件的 type: text 的 file-types 内加入了 conf，但它没有生效，我仍然无法编辑 nginx 的配置文件。 还有在添加驱动器的时候，驱动器的名称不能有数字零，不知道这是设计的如此还是 bug。

1. type: text 这段配置是用来配置缩略图生成规则的；关于可编辑的文件类型我再优化一下，支持一下在 admin 里配置支持编辑的格式
2. 名称不能包含 0，这个应该是个 bug，我修复一下

感谢~

**Comment by devld** [comment, marked as answer]
[`v0.6.5`](https://github.com/devld/go-drive/releases)

- 支持编辑的文件类型可以通过 `admin -> 站点` 进行配置了；
- 驱动名称不能包含 '0' 的问题也修复了
